src/hooks/useNodeJsContributorsApi.tsx

```diff
@@ -57,107 +67,107 @@ async function getMaxContributors(): Promise<[number, number]> {
  * @param randomPage
  */
 async function getContributor(randomPage: number): Promise<Contributor> {
-  const response = await fetch(`${CONTRIBUTORS_API_URI}&page=${randomPage}`);
-  const contributor = (await response.json())[0] as ContributorApiResponse;
-
-  return {
-    avatarUri: contributor.avatar_url,
-    commitsListUri: `https://github.com/nodejs/node/commits?author=${contributor.login}`,
-    contributionsCount: contributor.contributions,
-    login: contributor.login,
-    profileUri: contributor.html_url,
-  };
+  const response = await fetch(
+    `${CONTRIBUTORS_API_URI}&page=${randomPage}`
+  ).then(data => data.json() as Promise<ContributorApiResponse[]>);
+
+  const contributorData: Contributor[] = response.map(
+    ({ avatar_url, login, contributions, html_url }) => ({
+      avatarUri: avatar_url,
+      login,
+      contributionsCount: contributions,
+      profileUri: html_url,
+      commitsListUri: `https://github.com/nodejs/node/commits?author=${login}`,
+    })
+  );
+
+  const contributor = contributorData.shift() as Contributor;
+
+  if (window.localStorage) {
+    window.localStorage.setItem(
+      'contributors',
+      JSON.stringify(contributorData)
+    );
+  }
+
+  return contributor;
 }
 
 /**
  * Calls relative APIs and returns random contributor for Node.js main repo.
  * Trying to store cached data in localStorage in order to do less consequent requests
  */
-async function fetchRandomContributor() {
+export async function fetchRandomContributor(): Promise<Contributor> {
```

Comment:
Here's an improved version of this function that doesn't mix `async/await` and `.then` chaining. Also uses type inference. You don't need to do
`let thing: number | null = null`. Typescript will figure this out for you.
You also do not need the `new Promise` portion of the code

```suggestion
export async function fetchRandomContributor(): Promise<Contributor> {
  const ONE_MONTH_MS = 2592000000;

  let maxContributors = null;
  let fetchDate = null;
  let contributors: Contributor[] = [];
  if (window.localStorage) {
    maxContributors = parseInt(
      window.localStorage.getItem('max_contributors') || '',
      10
    );
    fetchDate = parseInt(
      window.localStorage.getItem('fetch_date') || '',
      10
    );
    contributors = JSON.parse(
      window.localStorage.getItem('contributors') || '[]'
    );
  }

  const needToRefetch = fetchDate && Date.now() - fetchDate >= ONE_MONTH_MS;

  if (contributors.length > 0 && !needToRefetch) {
    const contributor = contributors.shift();
    if (window.localStorage) {
      localStorage.setItem('contributors', JSON.stringify(contributors));
    }
    return contributor;
  }

  if (maxContributors && !needToRefetch) {
    return await getContributor(
      Math.floor(Math.random() * Math.floor(maxContributors)) + 1
    );
  }

  const [randomPage] = await getMaxContributors();
  const contributor = await getContributor(randomPage);
  if (window.localStorage) {
    window.localStorage.setItem('fetch_date', String(Date.now()));
  }
  return contributor;
}
```